Chapter06/recipe-04/c-example/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+# set minimum cmake version
+cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
+
+# project name and language
+project(recipe-04 C)
+
+# version of our project
+set(VERSION_MAJOR 2)
+set(VERSION_MINOR 0)
+set(VERSION_PATCH 1)
+
+# generate file version.h based on version.h.in
+configure_file(
+  ${PROJECT_SOURCE_DIR}/version.h.in
```

Comment:
Path of input file is relative to `CMAKE_CURRENT_SOURCE_DIR`, while path of output file is relative to `CMAKE_CURRENT_BINARY_DIR`. In this case giving the absolute path is redundant. I will explain it in the text for recipe 1, so consider removing the paths for consistency.